### PR TITLE
Add property for libvlc network-caching media option

### PR DESCRIPTION
**Pull request by john5788**
This changes the currently hard coded 100ms for libvlc's network-caching option to a user defined property. Without this change, an RTMP source sometimes "falls behind" causing the delay between real time and RTMP source to increase.

### CODE REVIEW

**Comment by ONSBalder** [comment]
Can one of the admins verify this patch?

**Comment by jp9000** [comment]
Did you test this code?

**Comment by john5788** [comment]
@jp9000 

I did test this and I have been running with my own patched version of vlc-video.so against 17.0.2. I discovered there was one missing line of code in my fork and I've updated to reflect this.

File: plugins/vlc-video/vlc-video-source.c [hunk 1]
```diff
@@ -417,9 +419,12 @@ static void add_file(struct vlc_source *c, struct darray *array,
 		new_media = create_media_from_file(path);
 
 	if (new_media) {
-		if (is_url)
-			libvlc_media_add_option_(new_media,
-					":network-caching=100");
+		if (is_url) {
+			struct dstr network_caching_option = {0};
+			dstr_catf(&network_caching_option, ":network-caching=%d", network_caching);
```
Comment (RytoEX) [inline plugins/vlc-video/vlc-video-source.c]: This line goes over the 80 character limit.  Insert a line break after "&network_caching_option," (right after the comma), delete the proceeding space character (right before the double quotation mark), and insert two tabs to indent.

File: plugins/vlc-video/vlc-video-source.c [hunk 2]
```diff
@@ -417,9 +419,12 @@ static void add_file(struct vlc_source *c, struct darray *array,
...
+			libvlc_media_add_option_(new_media, network_caching_option.array);
```
Comment (RytoEX) [inline plugins/vlc-video/vlc-video-source.c]: This line goes over the 80 character limit.  Insert a line break after "new_media," (right after the comma), delete the proceeding space character (right before "network_caching_option.array"), and insert two tabs to indent.

File: plugins/vlc-video/vlc-video-source.c [hunk 3]
```diff
@@ -520,13 +528,13 @@ static void vlcs_update(void *data, obs_data_t *settings)
 				dstr_copy(&dir_path, path);
 				dstr_cat_ch(&dir_path, '/');
 				dstr_cat(&dir_path, ent->d_name);
-				add_file(c, &new_files.da, dir_path.array);
+				add_file(c, &new_files.da, dir_path.array, network_caching);
```
Comment (RytoEX) [inline plugins/vlc-video/vlc-video-source.c]: This line goes over the 80 character limit. Insert a line break after "dir_path.array," (right after the comma), delete the proceeding space character (right before "network_caching"), and insert two tabs to indent.

File: plugins/vlc-video/vlc-video-source.c [hunk 4]
```diff
@@ -725,6 +733,8 @@ static obs_properties_t *vlcs_properties(void *data)
 	dstr_free(&filter);
 	dstr_free(&exts);
 
+	obs_properties_add_int(ppts, S_NETWORK_CACHING, T_NETWORK_CACHING, 100, 3600000, 10);
```
Comment (RytoEX) [inline plugins/vlc-video/vlc-video-source.c]: This line goes over the 80 character limit. Insert a line break after "100," (right after the comma), delete the proceeding space character (right before "3600000"), and insert two tabs to indent.

**Comment by EstevanTH** [comment]
This PR may have a conflict with my new patch #786. And I guess that with the ability to specify any media option, this PR is not really needed anymore. 😃

Just wanting to bring up good ideas, wouldn't it be more natural to add a field to `struct vlc_source` instead of using an additional argument to `static void add_file(...)`?

**Comment by jp9000** [comment]
I'm going to merge this PR over the other one, primarily because it's more simple and it's the only real value for VLC config that we really need.

Terribly sorry about the delay, last few months have been very busy and I had to put lower priority PRs on the backburner.

**Pull request closed without merging**